lib/widgets/message_list.dart

```diff
@@ -69,6 +70,26 @@ class MessageListAppBarTitle extends StatelessWidget {
 
   final Narrow narrow;
 
+  Widget _buildStreamRow(ZulipStream? stream, String text) {
+    final icon = switch (stream) {
+      ZulipStream(isWebPublic: true) => ZulipIcons.globe,
+      ZulipStream(inviteOnly: true) => ZulipIcons.lock,
+      ZulipStream() => ZulipIcons.hash_sign,
+      null => null, // A null [Icon.icon] makes a blank space.
+    };
+    return Row(
+      crossAxisAlignment: CrossAxisAlignment.baseline,
+      textBaseline: TextBaseline.alphabetic,
+      children: [
+        Padding(
+          // TODO(design): The vertical alignment of the stream privacy icon is ad hoc and eyeballed.
+          padding: const EdgeInsets.only(bottom: 4),
+          child: Icon(size: 16, icon)),
+        const SizedBox(width: 8),
+        Text(text),
+      ]);
+  }
+
   @override
   Widget build(BuildContext context) {
     switch (narrow) {
```

Comment:
The web app, in the heading of the views corresponding to `AllMessagesNarrow` and `DmNarrow`, these icons are used, respectively:

- <img width="21" alt="image" src="https://github.com/zulip/zulip-flutter/assets/22248748/71822ca6-0b59-4142-a0f4-51efcdba1a15">
- <img width="29" alt="image" src="https://github.com/zulip/zulip-flutter/assets/22248748/372b5668-1f1a-46e5-9c61-ab7683872fe1">

We could also use those.

In fact, I think the web app might have an icon for all possible message lists? Here are some others:

- <img width="23" alt="image" src="https://github.com/zulip/zulip-flutter/assets/22248748/4d90a8ac-9906-426c-9227-4f70e62e60f5">
- <img width="28" alt="image" src="https://github.com/zulip/zulip-flutter/assets/22248748/940391cb-8d9c-4b8f-b1b0-c3b224c07503">
- <img width="27" alt="image" src="https://github.com/zulip/zulip-flutter/assets/22248748/f71d541c-4254-48c0-8ebb-62c3f9a46c4e">

We don't support all message-list types that the web app does, but that might be helpful for choosing how to structure the code.

Reply:
Interesting, yeah.

It looks like at least some of those are FontAwesome icons, not Zulip custom icons. Here's what I see for "All messages":
`<i class="fa fa-align-left" aria-hidden="true"></i>`
So I think that's out of scope for this PR; I'll make a separate issue for bringing in FontAwesome icons. I might leave a blank space for it, though.

Reply:
#222.